plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeployImageHandler.java

```diff
@@ -35,15 +39,42 @@
 	@Override
 	public Object execute(ExecutionEvent event) throws ExecutionException {
 		DeployImageWizardModel model = new DeployImageWizardModel();
-		ISelection selection = HandlerUtil.getActivePart(event).getSite().getWorkbenchWindow().getSelectionService().getSelection();
+		IWorkbenchWindow window = HandlerUtil.getActivePart(event).getSite().getWorkbenchWindow();
+		ISelection selection = window.getSelectionService().getSelection();
+
+		boolean checkConnection = true;
 		final IDockerImage image = UIUtils.getFirstElement(selection, IDockerImage.class);
 		if(image != null) {
 			IDockerConnection dockerConnection = image.getConnection();
 			model.setOriginatedFromDockerExplorer(true);
 			model.setDockerConnection(dockerConnection);
 			model.setImageName(image.repo());
-		}else{
-			final IProject project = UIUtils.getFirstElement(selection, IProject.class);
+
+			checkConnection = false;
+			//Look for current selection in OpenShift Explorer
+			IViewPart part = window.getActivePage().findView(OpenShiftUIUtils.OPENSHIFT_EXPLORER_VIEW_ID);
+			if(part != null) {
+				selection = part.getSite().getSelectionProvider().getSelection();
+				if(selection != null && !selection.isEmpty()) {
+					checkConnection = true;
+				}
+			}
+		}
+		if(checkConnection) {
+			IProject project = UIUtils.getFirstElement(selection, IProject.class);
+			if(project == null) {
+				//If another resource is selected in OpenShift explorer, navigate to its project.
```

Comment:
if the 1st resource was selected in the properties view, the selection won't concern the OpenShift Explorer